arch/risc-v/src/common/riscv_macros.S

```diff
@@ -132,6 +202,59 @@
   REGLOAD    x30, REG_X30(\out)  /* t5 */
   REGLOAD    x31, REG_X31(\out)  /* t6 */
 
+#ifdef CONFIG_ARCH_FPU
+  REGLOAD      t0, REG_INT_CTX(\out)
```

Comment:
Having this inside the save/load fpu macro is fine as it only tampers with the saved context, not the status register directly